contracts/FlowIDTableStaking.cdc

```diff
@@ -857,6 +857,70 @@ pub contract FlowIDTableStaking {
             }
         }
 
+        /// Enforces slot (count) limits for each node type (role). If more nodes
+        /// of a given type are registered than the limit, they are removed
+        /// randomly until the limits are met. When a node is removed, it's
+        /// committed tokens are transfered to it's unstaked bucket.
+        pub fun slotSelect() {
+            let allNodeIDs = FlowIDTableStaking.getNodeIDs()
+
+            let limits = [2, 2, 2, 2, 2]
+            var counts = [0, 0, 0, 0, 0]
+            for nodeID in allNodeIDs {
+                let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
+                if nodeRecord.initialWeight == 100 {
+                    counts[nodeRecord.role - 1] = counts[nodeRecord.role - 1] + 1
+                }
+            }
+
+            for index, element in counts {
+                while counts[index] > limits[index] {
+                    let selection = unsafeRandom() % UInt64(counts[index])
```

Comment:
Same here: `element` instead of `counts[index]`

Reply:
This comment references the same line as the prior one. Was it meant to refer to a different line?

Reply:
I'll take another look